jedi/evaluate/docscrape_google.py

```diff
@@ -0,0 +1,308 @@
+        if len_ == 0:
+            # Empty lines take on their parents indentation
+            indent_ = prev_indent
+        true_indent.append(indent_)
+        prev_indent = indent_
+
+    # List of google style tags grouped by alias
+    tag_groups = [
+        ['Args', 'Arguments', 'Parameters', 'Other Parameters'],
+        ['Kwargs', 'Keyword Args', 'Keyword Arguments'],
+        ['Warns', 'Warning', 'Warnings'],
+        ['Returns', 'Return'],
+        ['Example', 'Examples'],
+        ['Note', 'Notes'],
+        ['Yields', 'Yield'],
+        ['Attributes'],
+        ['Methods'],
+        ['Raises'],
+        ['References'],
+        ['See Also'],
+        ['Todo'],
+    ]
+    # Map aliased tags to a cannonical name (the first item in the group).
+    tag_aliases = {item: group[0] for group in tag_groups for item in group}
+    tag_pattern = '^' + '(' + '|'.join(tag_aliases.keys()) + '): *$'
+
+    group_id = 0
+    prev_indent = 0
+    group_list = []
+    in_tag = False
+    for line_num, (line, indent_) in enumerate(zip(docstr_lines, true_indent)):
+        if re.match(tag_pattern, line):
+            # Check if we can look ahead
+            if line_num + 1 < len(docstr_lines):
+                # A tag is only valid if its next line is properly indented,
+                # empty, or is a tag itself.
+                indent_increase = true_indent[line_num + 1] > base_indent
+                indent_zero = line_len[line_num + 1] == 0
+                matches_tag = re.match(tag_pattern, docstr_lines[line_num + 1])
+                if (indent_increase or indent_zero or matches_tag):
+                    group_id += 1
+                    in_tag = True
+            else:
+                group_id += 1
+                in_tag = True
+        # If the indentation goes back to the base, then we have left the tag
+        elif in_tag and indent_ != prev_indent and indent_ == base_indent:
+            group_id += 1
+            in_tag = False
+        group_list.append(group_id)
+        prev_indent = indent_
+
+    # Group docstr lines by group list
+    groups_ = collections.defaultdict(list)
+    for groupid, line in zip(group_list, docstr_lines):
+        groups_[groupid].append(line)
+
+    groups = []
+    line_offset = 0
+    for k, lines in groups_.items():
+        if len(lines) == 0 or (len(lines) == 1 and len(lines[0]) == 0):
+            continue
+        elif len(lines) >= 1 and re.match(tag_pattern, lines[0]):
+            # An encoded google sub-block
+            key = lines[0].strip().rstrip(':')
+            val = lines[1:]
+            subblock = textwrap.dedent('\n'.join(val))
+        else:
+            # A top level text documentation block
+            key = '__DOC__'
+            val = lines[:]
+            subblock = '\n'.join(val)
+
+        key = tag_aliases.get(key, key)
+        groups.append((key, subblock))
+        line_offset += len(lines)
+    return groups
```

Comment:
The above function does the main work of splitting a docstring into blocks based on google-style headers.